### PR TITLE
Add bugprone-macro-parentheses clang-tidy check

### PR DESCRIPTION
**Pull request by gnawme**
Adds the [bugprone-macro-parentheses](https://clang.llvm.org/extra/clang-tidy/checks/bugprone/macro-parentheses.html) check to `clang-tidy`:

>Finds macros that can have unexpected behavior due to missing parentheses.

There are macros that use `std::stringstream` to simulate variadic functions; these should probably be converted to use actual parameter packs.

### CODE REVIEW

File: common/include/pcl/exceptions.h [hunk 1]
```diff
@@ -46,12 +46,14 @@
   * PCL_THROW_EXCEPTION(IOException,
   *                     "encountered an error while opening " << filename << " PCD file");
   */
+// NOLINTBEGIN(bugprone-macro-parentheses)
 #define PCL_THROW_EXCEPTION(ExceptionName, message)                         \
 {                                                                           \
   std::ostringstream s;                                                     \
-  s << message;                                                             \
+  s << message  ;                                                           \
```
Comment (mvieth) [inline common/include/pcl/exceptions.h:53]: Please undo this formatting change

File: search/include/pcl/search/impl/organized.hpp [hunk 2]
```diff
@@ -358,7 +358,7 @@ pcl::search::OrganizedNeighbor<PointT>::estimateProjectionMatrix ()
   }
 
   double residual_sqr = pcl::estimateProjectionMatrix<PointT> (input_, projection_matrix_, indices);
-  PCL_DEBUG_STREAM("[pcl::" << this->getName () << "::estimateProjectionMatrix] projection matrix=" << std::endl << projection_matrix_ << std::endl << "residual_sqr=" << residual_sqr << std::endl);
+  PCL_DEBUG_STREAM(std::string("[pcl::").append(this->getName ()) << std::string("::estimateProjectionMatrix] projection matrix=") << std::endl << projection_matrix_ << std::endl << std::string("residual_sqr=") << residual_sqr << std::endl);
```
Comment (mvieth) [inline search/include/pcl/search/impl/organized.hpp:361]: Sorry, I don't get why you did this?

**Review by mvieth — COMMENTED** [review]
Seems like there are also some parentheses needed in pcl_config.h (`PCL_VERSION_CALC`)

File: search/include/pcl/search/impl/organized.hpp (hunk 2 above)
Comment (gnawme) [reply search/include/pcl/search/impl/organized.hpp:361]: Sorry, I meant to revert this; I was trying to find a quick workaround for the pseudo-parameter-pack in the `PCL_DEBUG_STREAM` macro

**Comment by larshg** [comment]
The pcl_config.h is generated from https://github.com/PointCloudLibrary/pcl/blob/master/pcl_config.h.in
And you should remove the one from the build directory again 😀

File: ml/src/svm.cpp [hunk 3]
```diff
@@ -1908,7 +1909,7 @@ struct decision_function {
 static decision_function
 svm_train_one(const svm_problem* prob, const svm_parameter* param, double Cp, double Cn)
 {
-  double* alpha = Malloc(double, prob->l);
+  double* alpha = new double[prob->l];
```
Comment (mvieth) [inline ml/src/svm.cpp:1912]: Why did you change this?

Comment (gnawme) [reply ml/src/svm.cpp]: Sorry, didn't mean to commit

**Review by mvieth — APPROVED** [review]
Thanks!
